h2o-core/src/main/java/hex/grid/GridSearch.java

```diff
@@ -160,6 +160,61 @@ public long getModelCount() {
     return _hyperSpaceWalker.getMaxHyperSpaceSize();
   }
 
+
+  private class ModelFeeder<MP extends Model.Parameters> {
+
+    private final HyperSpaceWalker.HyperSpaceIterator<MP> hyperSpaceWalker;
+    private final Grid grid;
+    private final Lock lock = new ReentrantLock();
+
+    public ModelFeeder(HyperSpaceWalker.HyperSpaceIterator<MP> hyperSpaceWalker, Grid grid) {
+      this.hyperSpaceWalker = hyperSpaceWalker;
+      this.grid = grid;
+    }
+
+    private void feedModel(final Model model, final ParallelModelBuilder parallelModelBuilder) {
```

Comment:
Not cool. Really not cool, as it is responsible for manipulating internal state of the parallelModelBuilder, done this way only for simplicity.